chronograf.go

```diff
@@ -18,6 +18,41 @@ import (
 	"github.com/influxdata/influxdb/influxql"
 )
 
+// Chronograf User Roles
+const (
+	ViewerRoleName = "Viewer"
+	EditorRoleName = "Editor"
+	AdminRoleName  = "Admin"
+)
+
+// Default roles for chronograf Users
+var (
+	ViewerRole = Role{
+		Name: ViewerRoleName,
+	}
+
+	EditorRole = Role{
+		Name: EditorRoleName,
+	}
+
+	AdminRole = Role{
+		Name: AdminRoleName,
+	}
+)
+
+func RoleFromName(name string) (Role, error) {
```

Comment:
Validation of roles should be "close" to the data input.  This should be moved into the REST interface (/server).

Actually, I don't think that the new consts, vars nor `RoleFromName` should be here.

The store doesn't "care" about the name or if it is valid.  The piece of code that does care is the REST interface.

Reply:
This actually isn't a validation, it's just a lookup function for retrieving the correct Role from a string of that role's name.

Reply:
Yeah, I don't think there is a need for this function (nor the vars/consts really)

Reply:
The reason we did that is to prevent user-fragile string comparisons throughout the Roles pipeline, but we can change this.

Reply:
Checking for good/bad data should happen at the boundaries of the program.  In the case of chronograf that boundary is nearly always at the REST input/output.

More specifically to your point about user-fragile strings: it is not fragile to have code with assumptions. I'd say it is probably normal.  However, different layers of code should have strict responsibilities.  It is the responsibility of the REST interface to validate contextual data coming in and format it going out.